### ISSUE TITLE
Fencepost error in the old versions deletion script

### ISSUE THREAD

**Issue by 64kramsystem** [Tarvinder91/aws-utils#1]
Hi there!

Nice scripts :)

In the old versions deletion script, in the deletion cycle, a variable `next` is set.

The name of the variable is semantically incorrect; that's not the index of the first element of the next batch, instead, it's the index of the last element of the current batch.

Now, this wouldn't be a problem, but due to the semantical ambiguity, it does cause a bug - the jq array slicing uses this variable as the right end of the interval, but the right end is open, not close (see https://stedolan.github.io/jq/manual/#Basicfilters). See code [here](https://github.com/Tarvinder91/aws-utils/blob/ac9ff97e51f52dfa93e732b181f17bbb475090ad/S3/Delete-old-version-objects-1000obj-at-a-time.sh#L36).

In short:

```sh
    # this is the last element of the current batch
    next=$((i+999))

    # `jq .[$i:$next]` will _not_ include $next, that is, it will not include the last element of the current batch
    #
    oldversions=$(cat "old-objects-$bucket.json" |  jq '.[] | {Key,VersionId}' | jq -s '.' | jq .[$i:$next])

    # This is correct, as human interpretation is different from array slicing :)
    echo "Deleting records from $i - $next"
```

This also applies to the markers logic.

There's a chance I may be wrong, as it's the first time I work with jq, but I don't this this is the case.
